processor/transformprocessor/internal/common/functions.go

```diff
@@ -38,3 +40,11 @@ func Functions[K any]() map[string]interface{} {
 		"delete_matching_keys": ottlfuncs.DeleteMatchingKeys[K],
 	}
 }
+
+func ResourceFunctions() map[string]interface{} {
+	return Functions[ottlresource.TransformContext]()
+}
+
+func ScopeFunctions() map[string]interface{} {
+	return Functions[ottlscope.TransformContext]()
+}
```

Comment:
Why do you need these specialized calls?

Reply:
@bogdandrutu I made specialized calls to represent the possibility of Resource-only functions or Scope-only functions.

Reply:
Not sure we need a public API for that, just add a comment to the "Functions" that can be initialized with multiple contexts.

Reply:
This isn't a public API, it would never be moved to the OTTL.  This is a way for the transform processor to enforce its definition of what functions are allowed to be used for Scope and Resource.  Similar to how each signal in the transform processor has their own `functions.go`.